plugins/action_trim_responses/src/trimResponses.ts

```diff
@@ -73,15 +73,23 @@ const trimResponses: ActionPluginType = async ({ parameters, applicationData, DB
       (responseArray) => responseArray[responseArray.length - 1].id
     )
 
+    type Response = {
+      applicationResponseId: number
+      templateElementId: number
+    }
     // Run delete operation on all in toDelete array (new method)
-    const deletedResponses = reviewId
-      ? await db.deleteReviewResponses(responsesToDelete)
-      : await db.deleteApplicationResponses(responsesToDelete)
+    const deletedResponses = (
+      reviewId
+        ? await db.deleteReviewResponses(responsesToDelete)
+        : await db.deleteApplicationResponses(responsesToDelete)
+    ).sort((a: Response, b: Response) => a.applicationResponseId - b.applicationResponseId)
```

Comment:
I added the sorting here, as the array order was coming out different each time. As mentioned in the original comment, it'd be good to do more nuanced array matching (check elements but not order), but that's a bit more involved for now.